icu4c/source/test/intltest/rbbitst.cpp

```diff
@@ -5889,6 +5890,13 @@ void RBBITest::TestBug22602() {
     RuleBasedBreakIterator bi(rule, pe, ec);
 }
 
+void RBBITest::TestBug22636() {
+    UParseError pe {};
+    UErrorCode ec {U_ZERO_ERROR};
+    RuleBasedBreakIterator bi(u"A{77777777777777}", pe, ec);
```

Comment:
This test rule is missing its terminating semicolon, which causes the break iterator constructor to always return U_BRK_RULE_SYNTAX, no matter what the value of the integer tag.
```RuleBasedBreakIterator bi(u"A{2147483648};", pe, ec);```